PVTPackage/source/MultiphaseSystem/MultiphaseSystemProperties/BlackOilFlashMultiphaseSystemProperties.cpp

```diff
@@ -17,10 +17,12 @@
 namespace PVTPackage
 {
 
-BlackOilFlashMultiphaseSystemProperties::BlackOilFlashMultiphaseSystemProperties( std::size_t nComponents )
+BlackOilFlashMultiphaseSystemProperties::BlackOilFlashMultiphaseSystemProperties( std::vector< pvt::PHASE_TYPE > const & phases )
   :
-  BlackOilDeadOilMultiphaseSystemProperties( nComponents )
+  BlackOilDeadOilMultiphaseSystemProperties( phases )
 {
+  size_t const nComponents = phases.size();
```

Comment:
Could this line be factorized in `BlackOilDeadOilMultiphaseSystemProperties` as a protected member function
```c++
std::size_t getNComponents() const
{
    return getPhases().size();
}
```
With the appropriate physical explanations?

Reply:
Done, I added a getter with the physical explanations.